snippets/csharp/System.Security.Cryptography/Aes/Overview/program.cs

```diff
@@ -1,119 +1,81 @@
+        // Encrypt the string to an array of bytes.
+        var encrypted = EncryptStringToBytes(original, myAes.Key, myAes.IV);
 
-        //<Snippet3>
-        static string DecryptStringFromBytes_Aes(byte[] cipherText, byte[] Key, byte[] IV)
-        {
-            // Check arguments.
-            if (cipherText == null || cipherText.Length <= 0)
-                throw new ArgumentNullException("cipherText");
-            if (Key == null || Key.Length <= 0)
-                throw new ArgumentNullException("Key");
-            if (IV == null || IV.Length <= 0)
-                throw new ArgumentNullException("IV");
+        // Decrypt the bytes to a string.
+        var roundtrip = DecryptStringFromBytes(encrypted, myAes.Key, myAes.IV);
 
-            // Declare the string used to hold
-            // the decrypted text.
-            string plaintext = null;
-
-            // Create an Aes object
-            // with the specified key and IV.
-            using (Aes aesAlg = Aes.Create())
-            {
-                aesAlg.Key = Key;
-                aesAlg.IV = IV;
-
-                // Create a decryptor to perform the stream transform.
-                ICryptoTransform decryptor = aesAlg.CreateDecryptor(aesAlg.Key, aesAlg.IV);
-
-                // Create the streams used for decryption.
-                using (MemoryStream msDecrypt = new MemoryStream(cipherText))
-                {
-                    using (CryptoStream csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read))
-                    {
-                        using (StreamReader srDecrypt = new StreamReader(csDecrypt))
-                        {
-
-                            // Read the decrypted bytes from the decrypting stream
-                            // and place them in a string.
-                            plaintext = srDecrypt.ReadToEnd();
-                        }
-                    }
-                }
-            }
+        //Display the original data and the decrypted data.
+        Console.WriteLine("Original:   {0}", original);
+        Console.WriteLine("Round Trip: {0}", roundtrip);
+    }
+    
+    //<Snippet2>
+    static byte[] EncryptStringToBytes(string plainText, byte[] Key, byte[] IV)
+    {
+        // Create an Aes object
+        // with the specified key and IV.
+        using var aesAlg = Aes.Create();
+        aesAlg.Key = Key;
+        aesAlg.IV = IV;
+
+        // Create an encryptor to perform the stream transform.
+        var encryptor = aesAlg.CreateEncryptor(aesAlg.Key, aesAlg.IV);
+
+        // Create the streams used for encryption.
+        using var msEncrypt = new MemoryStream();
+        using var csEncrypt = new CryptoStream(msEncrypt, encryptor, CryptoStreamMode.Write);
+        using var swEncrypt = new StreamWriter(csEncrypt);
+        //Write all data to the stream.
+        swEncrypt.Write(plainText);
+        
+        
+        var encrypted = msEncrypt.ToArray();
```

Comment:
I don't think this is functionally the same code as it was before. Previously, `swEncrypt.Dispose()` was called before `msEncrypt.ToArray()` was called because of the explicit using block. This may have an impact on the flushing behavior of `swEncrypt`.

With the `using` as it is now,

Reply:
Latest commit should resolve this.